sdk/lib/src/imagekitio-angular/ik-image/ik-image.component.ts

```diff
@@ -46,39 +54,40 @@ export class IkImageComponent implements AfterViewInit, OnInit, OnChanges {
     }
   }
 
-  setUrl(src?, path?, transformation?, lqip?, urlEndpoint?, transformationPosition?, queryParameters?) {
-    const config = this.getConfigObject(src, path, transformation, transformationPosition, urlEndpoint, queryParameters)
+  setUrl(urlOptions: UrlOptions, lqip?:{active?:boolean, quality?:number}): void {
+    const config = this.getConfigObject(urlOptions);
     this.url = this.imagekit.getUrl(config);
     if (lqip && lqip.active === true) {
       this.lqipUrl = this.lqipload(lqip.quality, this.url, this.path);
     }
   }
 
-  getConfigObject(src?, path?, transformation?, transformationPosition?, urlEndpoint?, queryParameters?) {
-    const config = {
-      transformation: transformation,
+  getConfigObject(urlOptions: UrlOptions): Dict {
+    const config: Dict  = {
+      transformation : urlOptions.transformation
```

Comment:
@Aman, this is where the error is happening.
Transformation is a Partial<typeof supportedTransforms>
Compilation error is:
Type 'Partial<{ [key: string]: string; }>[]' is not assignable to type 'string | number | boolean | Partial<{ [key: string]: string; }> | QueryParameters'.